src/main/java/net/dv8tion/jda/internal/handle/ChannelUpdateHandler.java

```diff
@@ -216,46 +192,92 @@ private AbstractGuildChannelImpl<?> handleChannelTypeChange(AbstractGuildChannel
         EntityBuilder builder = getJDA().getEntityBuilder();
         GuildImpl guild = channel.getGuild();
 
-        if (newChannelType == ChannelType.TEXT)
-        {
-            //This assumes that if we're moving to a TextChannel that we're transitioning from a NewsChannel
-            NewsChannel newsChannel = (NewsChannel) channel;
-            getJDA().getNewsChannelView().remove(newsChannel.getIdLong());
-            guild.getNewsChannelView().remove(newsChannel.getIdLong());
-
-            TextChannelImpl textChannel = (TextChannelImpl) builder.createTextChannel(guild, content, guild.getIdLong());
+        ChannelType oldType = channel.getType();
 
-            //CHANNEL_UPDATE doesn't track last_message_id, so make sure to copy it over.
-            textChannel.setLatestMessageIdLong(newsChannel.getLatestMessageIdLong());
-
-            getJDA().handleEvent(
-                new ChannelUpdateTypeEvent(
-                    getJDA(), responseNumber,
-                    textChannel, ChannelType.NEWS, ChannelType.TEXT));
-
-            return textChannel;
+        long id = channel.getIdLong();
+        switch (oldType)
+        {
+        case TEXT:
+            getJDA().getTextChannelsView().remove(id);
+            guild.getTextChannelsView().remove(id);
+            break;
+        case NEWS:
+            getJDA().getNewsChannelView().remove(id);
```

Comment:
If there a way to dedupe some of this code between what's here and in `Guild#uncacheChannel`?